sdk/identity/azure-identity/inc/azure/identity/workload_identity_credential.hpp

```diff
@@ -27,15 +27,34 @@ namespace Azure { namespace Identity {
    */
   struct WorkloadIdentityCredentialOptions final : public Core::Credentials::TokenCredentialOptions
   {
+    /**
+     * @brief The TenantID of the service principal. Defaults to the value of the environment
+     * variable AZURE_TENANT_ID.
+     */
+    std::string TenantId;
```

Comment:
Instead of the logic for figuring out the values, do initialize the struct values with `= Core::_internal::Environment::GetVariable(...)`.
It is more consistent with the other SDKs, plus a customer can initialize a struct and see which values are being used.

Reply:
I don't think we should default the options to some value of the environment variable. I prefer using the empty string sentinel to mean the customer didn't set it.

This is [consistent with GoLang](https://github.com/Azure/azure-sdk-for-go/blob/70a663711035cf2dbddf5d65159cf4b40c636f19/sdk/azidentity/workload_identity.go#L34) and allows for an ordering of options which overrides the default:
1) User-defined option
2) Environment variable
3) Use default (for authority host) and throw for others

We could consider following .NET's approach on this, if we really see value on this. Let's discuss this separately.